### ISSUE TITLE
Questionable "piano" exemption in 1.3.4

### ISSUE THREAD

**Issue by anevins12** [w3c/wcag#3217]
The "piano" exemption in 1.3.4 is questionable:

- What type of piano was envisaged here, an 88 key grand piano? 
- Why is having an orientation lock more accessible than having an overflowing piano or small piano?
- What "landscape" was envisaged? A typical piano wouldn't even fit on a mobile phone landscape device.
- Why can't the piano have a smaller number of keys?
- Why can't octaves that would otherwise be hidden via overflow or too small to press be presented on a new row beneath?

We're talking about a digital representation of a piano here, we don't have to follow the constraints of physical limitations. You can chop up, reduce the number of keys, and improve the layout (without losing meaning) with a virtual piano.

**Comment by detlevhfischer** [comment]
>   What type of piano was envisaged here, an 88 key grand piano?

More like a toy piano with far fewer keys.

>    Why is having an orientation lock more accessible than having an overflowing piano or small piano?

A piano where you have to scroll to bring keys into the viewport is impractical, cannot be played continuously.

>    What "landscape" was envisaged? A typical piano wouldn't even fit on a mobile phone landscape device.

A reduced piano with say, two octaves and keys wider than, say, 44 CSS px would.

>    Why can't the piano have a smaller number of keys?

No one said it can't.

>    Why can't octaves that would otherwise be hidden via overflow or too small to press be presented on a new row beneath?

Its a possibility.

None of this takes away the main point that a thing like a piano keyboard is likely to require landscape orientation.

@anevins12  Can we close the issue?

**Comment by patrickhlauke** [comment]
I'll just say that "essential" here will depend on the intended use/activity. so yes, if it's intended to be played continuously and live, it's arguably only possible if as many keys as possible are shown simultaneously in a normal piano arrangement...

**Comment by anevins12** [comment]
Why would you have an orientation lock?

**Comment by anevins12** [comment]
You can win the case for a piano to be on landscape, but to say that locking orientation to landscape is a step further that needs explanation. Telling a person whose device orientation is fixed in their wheelchair that they're excluded from the piano is worse, in my view.

**Comment by patrickhlauke** [comment]
> You can win the case for a piano to be on landscape, but to say that locking orientation to landscape is a step further that needs explanation

can you explain what differentiation you're making here between "a piano to be on landscape" and "locking orientation to landscape"? to make sure the screen with the piano is displayed in landscape for the piano to be on landscape, you need to lock orientation of that screen to landscape, no?

**Comment by anevins12** [comment]
You could rotate the device to landscape without locking the orientation and throwing away the key, but the piano would have to be usable on portrait as well as landscape. Whereas the exception in this criterion allows people to exclude portrait users completely and show them a "oh you need to be using landscape screen, sorry not sorry" message.

**Comment by patrickhlauke** [comment]
> You could rotate the device to landscape without locking the orientation and throwing away the key, but the piano would have to be usable on portrait as well as landscape

this...does not make any sense. if the author has deemed it essential for the piano needing to be in landscape, saying "but the piano would have to be usable on portrait as well" is...basically saying the exception should not exist. so...

**Comment by anevins12** [comment]
I don't think the exception should exist, because I think looking at a piano, understanding its sound and messing around with it are valuable experiences that should be open to everyone.

**Comment by anevins12** [comment]
> edit: or are you saying there shouldn't be a doorslam "rotate your device" message? is THAT what you're getting at (and not the locking of orientation, but in fact NOT locking orientation). if so, that nuance wasn't clear here from the original issue

Yes

**Comment by anevins12** [comment]
I have now realized that "orientation lock" means something else.
